pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py

```diff
@@ -0,0 +1,418 @@
+
+        return True
+
+    def commercial_recovery(self, buildings, sample_damage_states, mcs_failure, redi_delay_factors, num_samples):
+        """
+        Calculates commercial building recovery for buildings
+
+        Args:
+            buildings(list): Buildings dataset
+            sample_damage_states (pd.DataFrame): Sample damage states
+            redi_delay_factors (pd.DataFrame): Delay factors based on REDi framework
+            mcs_failure (pd.DataFrame): Building inventory failure probabilities
+            num_samples (int): number of sample scenarios to use
+
+        Returns:
+            dict: dictionary with id/guid and commercial recovery for each quarter
+
+        """
+
+        start_total_delay = time.process_time()
+        total_delay = CommercialBuildingRecovery.total_delay(buildings, sample_damage_states, mcs_failure, redi_delay_factors, num_samples)
+        end_total_delay = time.process_time()
+        print("Finished executing total_delay() in " + str(end_total_delay - start_total_delay) + " secs")
+
+        recovery = self.recovery_rate(buildings, sample_damage_states, total_delay)
+        end_recovery = time.process_time()
+        print("Finished executing recovery_rate() in " + str(end_recovery - end_total_delay) + " secs")
+
+        time_stepping_recovery = CommercialBuildingRecovery.time_stepping_recovery(recovery)
+        end_time_stepping_recovery = time.process_time()
+        print("Finished executing time_stepping_recovery() in " +
+              str(end_time_stepping_recovery - end_recovery) + " secs")
+
+        #result = time_stepping_recovery
+
+        end_time = time.process_time()
+        print("Analysis completed in " + str(end_time - start_total_delay) + " secs")
+
+        return total_delay, recovery, time_stepping_recovery
+
+    @staticmethod
+    def total_delay(buildings, sample_damage_states, mcs_failure, redi_delay_factors, num_samples):
+        """ Calculates total delay by combining financial delay and other factors from REDi framework
+
+        Args:
+            buildings (list): List of buildings
+            sample_damage_states (pd.DataFrame): Building inventory damage states.
+            mcs_failure (pd.DataFrame): Building inventory failure probabilities
+            redi_delay_factors (pd.DataFrame): Delay impeding factors such as post-disaster inspection, insurance claim,
+                financing, and government permit based on building's damage state.
+            num_samples (int): number of sample scenarios to use
+
+        Returns:
+            pd.DataFrame: Total delay time of all impeding factors from REDi framework.
+        """
+
+        # Obtain the commercial buildings in damage
+        damage = mcs_failure[mcs_failure['haz_expose'] == 'yes']
+        commercial = []
+        for i, b in enumerate(buildings):
+            if b['properties']['archetype'] in (6, 7, 8, 15, 16, 18, 19):
```

Comment:
the archetypes are hardcoded here. is there a better way? Maybe pass in that as a parameter?